### PR TITLE
Forgot Password

### LINKED ISSUE #3: Forgot Password
Requests 1
* POST (for forgot password)
* sends it an email
* it finds the user in the db, adds a pin db, adds expiration date
*actually sends an email with the pin

Requests 2 
* POST request
* pass in the email, pin, new password
* validate the pin with expiry date
* update the password
* send you the jwt token

### PR DESCRIPTION
**Pull request by josh-byster**
Fixes #3

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,15 +1,18 @@
...
+var bodyParser = require('body-parser')
 const app = express();
+
+app.use(bodyParser.urlencoded())
 app.use(cors());
 app.use(bodyParser.json());
 app.use(bodyParser.urlencoded({ extended: true }));
 
+const sgMail = require('@sendgrid/mail');
+sgMail.setApiKey(process.env.SENDGRID_API_KEY);
```
Comment (CodeBrew28) [inline src/App.js]: are we still using this or can this be remove?

File: docs/features/forgot-password.mdx [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+
+12. If all is successful, expiration is invalidated and password is changed. If not (email doesn't match, PIN is expired, etc.) then an error message will be returned to the client.
+User gets logged in with the returned JWT.
+
+***Security Question? Make it an optional feature before the pin***
+
+* Set a default security question in the database (if the user doesnt exist)
+* Field with every user for specific questions
+* Dev team can choose questions, auth server handles everything. 
+* Config file can turn off this functionality
```
Comment (CodeBrew28) [inline docs/features/forgot-password.mdx]: all of this looks good to me :)

**Comment by HRimaw2** [comment]
Seems good just needs to pass now bot

File: src/App.js [hunk 3]
```diff
@@ -26,26 +27,147 @@ app.get("/users", async function(req, res) {
...
+    res.send({
+      status: 400,
+      message: "PIN is expired"
+    }); 
+    return;
+  } 
+  //user matches, change expiration
+  var date = new Date();
+  // remove a day to the current date to expire it
+  date.setDate(date.getDate() - 1);
```
Comment (CodeBrew28) [inline src/App.js]: add a comment explaining it

File: src/App.js [hunk 4]
```diff
@@ -1,11 +1,12 @@
 const express = require("express");
+const nodemailer = require("nodemailer");
 const mongoose = require("mongoose");
 const cors = require("cors");
-const bodyParser = require("body-parser");
-
 const User = require("./models/User");
-
+var bodyParser = require("body-parser");
```
Comment (CodeBrew28) [inline src/App.js]: make it a const

File: src/models/User.js [hunk 5]
```diff
@@ -1,8 +1,16 @@
 const mongoose = require("mongoose");
-const uri = "mongodb://product:infra28@ds125385.mlab.com:25385/auth-infra"; 
-const db = mongoose.createConnection(uri);
+const uri = "mongodb://product:infra28@ds125385.mlab.com:25385/auth-infra";
+const db = mongoose.connect(uri, { useNewUrlParser: true });
```
Comment (CodeBrew28) [inline src/models/User.js]: yeet

**Pull request merged**